bgpd/bgp_fsm.c

```diff
@@ -1892,6 +1892,10 @@ static enum bgp_fsm_state_progress bgp_start(struct peer_connection *connection)
 				 __func__, peer->connection->fd);
 			return BGP_FSM_FAILURE;
 		}
+		/* after connect is called(), getpeername is able to return
+		 * port and address on non established streams
+		 */
+		bgp_updatesockname(peer);
```

Comment:
Why not adopt `bgp_getsockname()` for this purpose and reuse it?

Reply:
I made a proposal like you suggest.